server/chat-plugins/responder.ts

```diff
@@ -0,0 +1,517 @@
+/**
+ * PS Help room auto-response plugin.
+ * Uses Regex to match room frequently asked question (RFAQ) entries,
+ * and replies if a match is found.
+ * Supports configuration, and works in all rooms, though intended mainly for Help.
+ * Written by Mia.
+ * @author mia-pi-git
+ */
+
+import {FS} from '../../lib/fs';
+import {Utils} from '../../lib/utils';
+import {LogViewer} from './chatlog';
+import {roomFaqs} from './room-faqs';
+
+const PATH = 'config/chat-plugins/responder.json';
+// 4: filters out conveniently short aliases
+const MINIMUM_LENGTH = 4;
+
+export let answererData: {[roomid: string]: PluginData} = {};
+
+try {
+	answererData = JSON.parse(FS(PATH).readSync());
+} catch (e) {}
+
+function loadOldData() {
+	const oldDataPath = 'config/chat-plugins/help.json';
+	const oldPath = FS(oldDataPath);
+	if (oldPath.existsSync() && Config.helpFilterRoom) {
+		const oldData = JSON.parse(oldPath.readSync());
+		answererData[toID(Config.helpFilterRoom)] = oldData;
+		oldPath.unlinkIfExistsSync();
```

Comment:
Maybe wait to make sure it works correctly on the live server?

Reply:
I thoroughly tested it locally. I can remove the unlink, though.

Reply:
Removed the unlink. Now it'll check it, but only set the current data to the old data if it hasn't been changed since.